packages/batch/src/BasePartialProcessor.ts

```diff
@@ -66,13 +74,45 @@ abstract class BasePartialProcessor {
    * Call instance's handler for each record
    * @returns List of processed records
    */
-  public async process(): Promise<(SuccessResponse | FailureResponse)[]> {
+  public process(): (SuccessResponse | FailureResponse)[] {
+    /**
+     * If this is an async processor, user should have called processAsync instead,
+     * so we call the method early to throw the error early thus failing fast.
+     */
+    if (this.constructor.name === 'AsyncBatchProcessor') {
+      this.processRecord(this.records[0]);
+    }
     this.prepare();
 
     const processedRecords: (SuccessResponse | FailureResponse)[] = [];
     for (const record of this.records) {
-      processedRecords.push(await this.processRecord(record));
+      processedRecords.push(this.processRecord(record));
+    }
+
+    this.clean();
+
+    return processedRecords;
+  }
+
+  /**
+   * Call instance's handler for each record
+   * @returns List of processed records
+   */
+  public async processAsync(): Promise<(SuccessResponse | FailureResponse)[]> {
```

Comment:
Should this be named to `asyncProcess()` to be consistent with the method name `asyncProcessRecord()`?

Reply:
I wasn't sure at first but I think you are right, I have changed it.